core/services/ocr2/plugins/ocr2keeper/evmregistry/v20/registry.go

```diff
@@ -103,6 +102,9 @@ func NewEVMRegistryService(addr common.Address, client legacyevm.Chain, lggr log
 		enc:      EVMAutomationEncoder20{},
 	}
 
+	r.ctx, r.cancel = context.WithCancel(context.Background())
```

Comment:
Apologies if this has been talked about before, but why are we ok with a background context in this case?

Reply:
This is a side effect of our Start/Close pattern. We have to create new contexts somewhere, although storing a context on a field is not recommended. Usually we have a stopChan, and then derive new contexts from it.

Reply:
Is storing context on a field not recommended because we end up plumbing context through the fields (r.ctx, and r.cancel) and the method arguments (i.e. `CheckUpkeep`)?